Lib/ufo2ft/outlineCompiler.py

```diff
@@ -584,55 +584,100 @@ def setupTable_hmtx(self):
+            "caretOffset": "%sCaretOffset" % commonPrefix,
+        }
+        for otfName, ufoName in metricsDict.items():
+            setattr(table, otfName,
+                    getAttrWithFallback(font.info, ufoName))
+        # Horizontal metrics in hhea, vertical metrics in vhea
+        advances = []  # width in hhea, height in vhea
+        firstSideBearings = []  # left in hhea, top in vhea
+        secondSideBearings = []  # right in hhea, bottom in vhea
         extents = []
         for glyphName in self.allGlyphs:
-            width, left = hmtx[glyphName]
-            widths.append(width)
+            advance, firstSideBearing = mtxTable[glyphName]
+            advances.append(advance)
             bounds = self.glyphBoundingBoxes[glyphName]
             if bounds is None:
                 continue
-            right = width - left - (bounds.xMax - bounds.xMin)
-            lefts.append(left)
-            rights.append(right)
+            if isHhea:
+                boundsAdvance = (bounds.xMax - bounds.xMin)
+            else:
+                boundsAdvance = (bounds.yMax - bounds.yMin)
+            secondSideBearing = advance - firstSideBearing - boundsAdvance
+            firstSideBearings.append(firstSideBearing)
+            secondSideBearings.append(secondSideBearing)
             # equation from the hhea spec for calculating xMaxExtent:
             #   Max(lsb + (xMax - xMin))
-            extent = left + (bounds.xMax - bounds.xMin)
+            extent = firstSideBearing + boundsAdvance
             extents.append(extent)
-        hhea.advanceWidthMax = max(widths)
-        hhea.minLeftSideBearing = min(lefts) if lefts else 0
-        hhea.minRightSideBearing = min(rights) if rights else 0
-        hhea.xMaxExtent = max(extents) if extents else 0
-        # misc
-        hhea.caretSlopeRise = getAttrWithFallback(font.info, "openTypeHheaCaretSlopeRise")
-        hhea.caretSlopeRun = getAttrWithFallback(font.info, "openTypeHheaCaretSlopeRun")
-        hhea.caretOffset = round(getAttrWithFallback(font.info, "openTypeHheaCaretOffset"))
-        hhea.reserved0 = 0
-        hhea.reserved1 = 0
-        hhea.reserved2 = 0
-        hhea.reserved3 = 0
-        hhea.metricDataFormat = 0
+        setattr(table,
+                "advance%sMax" % ("Width" if isHhea else "Height"),
+                max(advances) if advances else 0)
+        setattr(table,
+                "min%sSideBearing" % ("Left" if isHhea else "Top"),
+                min(firstSideBearings) if firstSideBearings else 0)
+        setattr(table,
+                "min%sSideBearing" % ("Right" if isHhea else "Bottom"),
+                min(secondSideBearings) if secondSideBearings else 0)
+        setattr(table,
+                "%sMaxExtent" % ("x" if isHhea else "y"),
+                max(extents) if extents else 0)
+        if isHhea:
+            reserved = range(4)
+        else:
+            reserved = range(5)
```

Comment:
`table__v_h_e_a.reserved0` is deprecated and now it is an alias for `table__v_h_e_a.caretOffset` (see 
[_v_h_e_a.py#L110-L117](https://github.com/fonttools/fonttools/blob/3.24.1/Lib/fontTools/ttLib/tables/_v_h_e_a.py#L110-L117)). So this line should be `reserved = range(1, 5)`.

Reply:
Indeed! Thanks!